presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -286,12 +294,21 @@ private int readWithFilter(int[] positions, int positionCount)
     private byte evaluateFilter(int position, int index, int length)
     {
         if (filter.testLength(length)) {
-            int currentLength = dictionary.getSliceLength(index);
-            Slice data = dictionary.getSlice(index, 0, currentLength);
+            int currentLength = currentDictionaryOffsetVector[index + 1] - currentDictionaryOffsetVector[index];
```

Comment:
This logic needs some refactoring in order to be easier to read. We are making 3 different tests here: 
1. filter.testLength(length)
2. filter.testBytes(valueWithPadding, 0, length); in the case of charType
3. filter.testBytes(currentDictionaryData, currentDictionaryOffsetVector[index], length); in case of non-charType

If any of these filters fail, we return FILTER_FAILED and in case of pass we return FILTER_PASSED. Using this information we can flatten this logic as follows:
```
if (!filter.testLength(length))
    return FILTER_FAILED;
if (isCharType && length != currentLength) {
    //...
    if (!filter.testBytes(valueWithPadding, 0, length))
        return FILTER_FAILED;
} else if (!filter.testBytes(currentDictionaryData, currentDictionaryOffsetVector[index], length)) {
    return FILTER_FAILED;
}

// the stuff you do once if(pass) {...}.

```

Now we don't need the pass state variable and branches are clearer.